src/core/services/shift_service.py

```diff
@@ -59,7 +61,20 @@ async def get_shift(self, id: UUID) -> Shift:
         return await self.__shift_repository.get(id)
 
     async def update_shift(self, id: UUID, update_shift_data: ShiftUpdateRequest) -> Shift:
-        return await self.__shift_repository.update(id, Shift(**update_shift_data.dict(exclude_unset=True)))
+        shift: Shift = await self.__shift_repository.get(id)
+        if shift.status in (Shift.Status.CANCELLED, Shift.Status.FINISHED):
+            if (
+                shift.started_at != update_shift_data.started_at.date()
+                or shift.finished_at != update_shift_data.finished_at.date()
+                or shift.final_message != update_shift_data.final_message
+            ):
+                raise ShiftUpdateException(detail="В закончившейся или отмененной смене можно изменить только название")
+        if shift.status == Shift.Status.STARTED:
+            if shift.started_at != update_shift_data.started_at.date():
+                raise ShiftUpdateException(detail="Нельзя изменить дату начала текущей смены")
+            if update_shift_data.finished_at.date() < date.today():
+                raise ShiftUpdateException(detail="Нельзя установить дату окончания текущей смены прошедшим числом")
+        return await self.__shift_repository.update(id, Shift(**update_shift_data.dict()))

```

Comment:
Вся же эта переделка затеивалась для того, чтобы избавиться от этого 💩 `Shift(**update_shift_data.dict())`.
Добавь, пожалуйста, замечания об этом в кодстайл нашей ВиКи и сделай пост в общем канале, что так сохранять код нельзя. Только передавая полный объект. И попроси всех перечитать эту ВиКи, там много чего полезного!
```suggestion
        return await self.__shift_repository.update(id, shift)
```

Reply:
Кратко написал: https://www.notion.so/codestyle-fc03c29c1edf4df7a335727efb5cc075#791ceeb0a8b44bd0afab41685a8ebe5a

Reply:
Исправил.